service/internal/nodecount/node_count.go

```diff
@@ -132,14 +138,15 @@ func (nc *NodeCount) cachedNodes(ctx context.Context, cr metav1.Object) (corev1.
 	return nodes, nil
 }
 
-func (nc *NodeCount) lookupNodes(ctx context.Context) (corev1.NodeList, error) {
-	// TODO we need to get rid off the controllercontext.Context but for now it should be fine.
-	cc, err := controllercontext.FromContext(ctx)
-	if err != nil {
+func (nc *NodeCount) lookupNodes(ctx context.Context, cr metav1.Object) (corev1.NodeList, error) {
+	client, err := nc.tenantClient.K8sClient(ctx, cr)
+	if tenantclient.IsNotAvailable(err) {
+		return corev1.NodeList{}, microerror.Mask(err)
```

Comment:
The result returned is the same as below so the condition does not seem to serve a real purpose. I think we can simply forward the error we receive and that is it.